Initial commit of new TrailingStopLossIndicator with test class.

ElMoe opened the pull request:
For details see source code.
For feature discussion see: https://github.com/mdeverdelhan/ta4j/issues/61

mdeverdelhan commented:
Thank you! I'm going to merge this and make some cosmetic changes.

mdeverdelhan commented:
Here the changes I made:
  * I used the `Decimal#isGreaterThan(Decimal)` method instead of a raw compareTo
  * I changed the license header and added you [there](https://github.com/mdeverdelhan/ta4j/blob/master/AUTHORS)
  * I fixed some typo
  * I refactored names to better match the [Java code conventions](http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-135099.html#367)

I plan to:
  * Update your test class (to simplify it): less decimals for raw numbers mainly
  * Fix the without-initial-limit-argument case (Now you compute `initialStopLossLimit` in the constructor but the provided indicator may not have any value yet. It should be initialized with `Decimal.NaN` and computed in your first `calculate(int)` call.)

ElMoe commented:
Great, thanks for merging!

> Am 08.01.2016 um 10:05 schrieb Marc de Verdelhan <notifications@github.com>:
> 
> Here the changes I made:
> 
> I used the Decimal#isGreaterThan(Decimal) method instead of a raw compareTo
> I changed the license header and added you there <https://github.com/mdeverdelhan/ta4j/blob/master/AUTHORS>
> I fixed some typo
> I refactored names to better match the Java code conventions <http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-135099.html#367>
> I plan to:
> 
> Update your test class (to simplify it): less decimals for raw numbers mainly
> Fix the without-initial-limit-argument case (Now you compute initialStopLossLimit in the constructor but the provided indicator may not have any value yet. It should be initialized with Decimal.NaN and computed in your first calculate(int) call.)
> —
> Reply to this email directly or view it on GitHub <https://github.com/mdeverdelhan/ta4j/pull/63#issuecomment-169940089>.
>
